portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -54,5 +56,43 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       response.sendRedirect(loginUrl);
     }
   }
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String nickname = getParameter(request, "nickname", "");
+    String id = userService.getCurrentUser().getUserId();
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity entity = new Entity("UserInfo", id);
+    entity.setProperty("id", id);
+    entity.setProperty("nickname", nickname);
+    
+    datastore.put(entity);
+
+    response.sendRedirect("/index.html");
+  }
 
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```

Comment:
Consider using java.util.Optional for this. It will give more flexibility and is more readable.

```java
private Optional<String> getParameter(HttpServletRequest request, String name) {
  return Optional.ofNullable(request.getParameter(name));
}
```

then the caller can specify the default

```java
String nickname = getParameter(request, "nickname").orElse("");
```

or even handle it differently

```java
getParameter(request, "nickname").ifPresent(nickname => {
  ...
});
```

Reply:
Thanks for the edits. That looks interesting. Changes made.

Reply:
It threw an this exception.
https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PreparedQuery.TooManyResultsException

Reply:
That error seems unrelated to this suggestion. That error seems to indicate that the Query expected to find a single entry but instead found multiple.

Reply:
I reverted whatever changes I made to the state when everything was working and made this specific change and I got that error.

Reply:
Yes, I think this change had to do nothing with that exception. The query did return multiple results because there were multiple entities having the same id (userInfo id, same user with multiple nickname)